ARMeilleure/Translation/Translator.cs

```
@@ -368,29 +393,46 @@ private static ControlFlowGraph EmitAndGetCFG(ArmEmitterContext context, Block[]
             return context.GetControlFlowGraph();
         }
 
-        internal static void EmitSynchronization(EmitterContext context)
+        internal static void EmitRejitCheck(ArmEmitterContext context, out Counter<uint> counter)
         {
-            long countOffs = NativeContext.GetCounterOffset();
+            const int MinsCallForRejit = 100;
 
-            Operand countAddr = context.Add(context.LoadArgument(OperandType.I64, 0), Const(countOffs));
+            if (!Counter<uint>.TryCreate(context.CountTable, out counter))
+            {
+                return;
+            }
```

I think we should raise an exception here, in the hypothetical case of not being able to allocate a further entry in the count table, also to avoid not being able to have rejits. In this way we can realise the limit and increase the capacity of the table.